streaming/src/main/scala/org/apache/spark/streaming/dstream/DStream.scala

```diff
@@ -906,12 +908,16 @@ abstract class DStream[T: ClassTag] (
   /**
    * Save each RDD in this DStream as at text file, using string representation
    * of elements. The file name at each batch interval is generated based on
-   * `prefix` and `suffix`: "prefix-TIME_IN_MS.suffix".
+   * `prefix` and `suffix`: "prefix-TIME_IN_MS.suffix". If the `CompressionCodec`
+   * is defined, it will use specific `CompressionCodec` to compress the text.
    */
-  def saveAsTextFiles(prefix: String, suffix: String = ""): Unit = ssc.withScope {
```

Comment:
This ends up changing the API at a binary level even, right? I don't think you can do this. As I say, I am not sure the intent is to keep augmenting the `DStream` API where it duplicates the `RDD` API